tests/unit/fem/test_cmplx_gridfunc_save.cpp

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) 2010-2025, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "mfem.hpp"
+#include "unit_tests.hpp"
+
+using namespace mfem;
+
+TEST_CASE("ComplexGridFunction Save", "[ComplexGridFunction]")
+{
+   const int order = 3;
+
+   Mesh mesh = Mesh::MakeCartesian3D(1, 2, 3, Element::PYRAMID,
+                                     1.0, 2.0, 3.0);
+
+   H1_FECollection fec_h1(order, mesh.Dimension());
+   ND_FECollection fec_nd(order, mesh.Dimension());
+   RT_FECollection fec_rt(order-1, mesh.Dimension());
+   L2_FECollection fec_l2(order-1, mesh.Dimension());
+
+   FiniteElementSpace fes_h1(&mesh, &fec_h1);
+   FiniteElementSpace fes_nd(&mesh, &fec_nd);
+   FiniteElementSpace fes_rt(&mesh, &fec_rt);
+   FiniteElementSpace fes_l2(&mesh, &fec_l2);
+
+   ComplexGridFunction gf_h1(&fes_h1);
+   ComplexGridFunction gf_nd(&fes_nd);
+   ComplexGridFunction gf_rt(&fes_rt);
+   ComplexGridFunction gf_l2(&fes_l2);
+
+   gf_h1.Randomize(1);
+   gf_nd.Randomize(1);
+   gf_rt.Randomize(1);
+   gf_l2.Randomize(1);
+
+   Vector zeroVec(3); zeroVec = 0.0;
+   ConstantCoefficient zeroCoef(0.0);
+   VectorConstantCoefficient zeroVecCoef(zeroVec);
+
+   const double norm_h1 = gf_h1.ComputeL2Error(zeroCoef, zeroCoef);
+   const double norm_nd = gf_nd.ComputeL2Error(zeroVecCoef, zeroVecCoef);
+   const double norm_rt = gf_rt.ComputeL2Error(zeroVecCoef, zeroVecCoef);
+   const double norm_l2 = gf_l2.ComputeL2Error(zeroCoef, zeroCoef);
+
+   std::ofstream ofs_h1("cmplx_h1.gf"); ofs_h1.precision(8);
+   std::ofstream ofs_nd("cmplx_nd.gf"); ofs_nd.precision(8);
+   std::ofstream ofs_rt("cmplx_rt.gf"); ofs_rt.precision(8);
+   std::ofstream ofs_l2("cmplx_l2.gf"); ofs_l2.precision(8);
+
+   gf_h1.Save(ofs_h1);
+   gf_nd.Save(ofs_nd);
+   gf_rt.Save(ofs_rt);
+   gf_l2.Save(ofs_l2);
+
```

Comment:
The same as below 👇 here with `ofstream`s. You may use the other variant of `Save()` to do it internally. It also prevents starting the load procedure before the save is done.

Reply:
You forgot these 😊

Reply:
Yes, I did. Thanks